Add problem task module and info module

mdolin opened the pull request.

tadeboro commented on plugins/modules/problem_task.py:
```diff
@@ -0,0 +1,435 @@
+
+extends_documentation_fragment:
+  - servicenow.itsm.instance
+  - servicenow.itsm.sys_id.info
+  - servicenow.itsm.number.info
```
Are these two OK? I would assume we need regular ones here since we are not in the info module.

```suggestion
  - servicenow.itsm.sys_id
  - servicenow.itsm.number
```

tadeboro commented on plugins/modules/problem_task.py:
```diff
@@ -0,0 +1,435 @@
+  - servicenow.itsm.number.info
+
+seealso:
+  - module: servicenow.itsm.problem_task_info
```
Maybe we should also link to problem modules here?

tadeboro commented on plugins/modules/problem_task_info.py:
```diff
@@ -0,0 +1,228 @@
+  - servicenow.itsm.number.info
+  - servicenow.itsm.query
+seealso:
+  - module: servicenow.itsm.problem_task
```
Maybe add references to regular problem modules?

tadeboro reviewed (changes requested):
Almost there. I just found a few nitpicks. Once those are settled, we should be good to go.

As for the problematic state change, I did not find anything that would pop out right now, so we will handle that another time and with some help from the snow experts if needed).

Commit:
Update table module util to find the problem

Implements method for finding the problem which
is then used and assigned in the problem task module.

Commit:
Implements ServiceNow problem task module

In the current implementation, basic functionalities
creating and deleting problem tasks are covered.
We also present often used fields that are important
for problem tasks to be created or in any other
way manipulated. For all other less used or optional
parameters, they can be strung together under other
field in the problem tasks as they are.

There is an issue with the state transition from
Assess to Work In Progress and to Closed.
When manually transitioning to Work In Progress state
fields started_at and stared_by are field in but
the module can not change to that state even though
these two parameters are filled.

Commit:
Add reference to problem task modules

tadeboro reviewed (approved):
Thank you, @mdolin, for your contribution!
